wagtail/core/management/commands/create_log_entries_from_revisions.py

```diff
@@ -0,0 +1,49 @@
+from django.core.management.base import BaseCommand
+from wagtail.core.models import PageLogEntry, PageRevision
+
+
+def get_comparison(page, revision_a, revision_b):
+    comparison = page.get_edit_handler().get_comparison()
+    comparison = [comp(revision_a, revision_b) for comp in comparison]
+    comparison = [comp for comp in comparison if comp.has_changed()]
+
+    return comparison
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        current_page_id = None
+        previous_revision = None
+        for revision in PageRevision.objects.order_by('page_id', 'created_at').select_related('page').iterator():
+            is_new_page = revision.page_id != current_page_id
+            has_content_changes = False
+            current_page_id = revision.page_id
+
+            current_revision_as_page = revision.as_page_object()
+
+            if not PageLogEntry.objects.filter(revision=revision).exists():
+                published = revision.id == revision.page.live_revision_id
```

Comment:
We did have some code that detected past published revisions by looking at what the `live_revision_id` was in the `content_json` field. Did we remove that?

Reply:
We did - https://gist.github.com/zerolab/31605438a6ac72dcf64d24e320ea0247#file-audit-log-original-diff-L381 - it was setting the plublished flag to true at the end of the process, however we dropped the flag, so no need.

There's additional logic (around `if is_new_page or has_content_changes or published`) that deals with published revisions and it is a simpler process.

Reply:
Ahh, yep. We are now adding an additional event for published revisions aren't we? Do we want to have a full history of publish events?

Reply:
sorted (and fixed the comparison too)